crates/bevy_ecs/src/system/system_registry.rs

```diff
@@ -500,6 +500,16 @@ impl World {
 
     /// Runs a cached system, registering it if necessary.
     ///
+    /// # Type Inference Note
+    /// If the system returns `()`, you may need to explicitly constrain the output
+    /// type for error handling:
+    ///
+    /// ```rust
```

Comment:
You'll need to declare `world` and `my_system` for the doc tests to compile, like

```suggestion
    /// ```rust
    /// # use bevy_ecs::prelude::*;
    /// # fn my_system() -> Result { Ok(()) }
    /// # let mut world = World::new();
```